### PR TITLE
refactor(editor): Convert Tooltip to composition api

### PR DESCRIPTION
**Pull request by tomi**
## Summary

Convert Tooltip to composition api

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/design-system/src/components/N8nTooltip/Tooltip.vue [hunk 1]
```diff
@@ -48,15 +37,15 @@ export default defineComponent({
 		<slot />
 		<template #content>
 			<slot name="content">
-				<div v-html="content"></div>
+				<div v-html="props.content"></div>
```
Comment (alexgrozav) [inline packages/design-system/src/components/N8nTooltip/Tooltip.vue:40]: You should omit `props` in `<template>`.

```suggestion
				<div v-html="content"></div>
```

File: packages/design-system/src/components/N8nTooltip/Tooltip.vue [hunk 2]
```diff
@@ -48,15 +37,15 @@ export default defineComponent({
...
 			</slot>
 			<div
-				v-if="buttons.length"
+				v-if="props.buttons.length"
```
Comment (alexgrozav) [inline packages/design-system/src/components/N8nTooltip/Tooltip.vue:43]:
```suggestion
				v-if="buttons.length"
```

File: packages/design-system/src/components/N8nTooltip/Tooltip.vue [hunk 3]
```diff
@@ -48,15 +37,15 @@ export default defineComponent({
...
 				:class="$style.buttons"
-				:style="{ justifyContent: justifyButtons }"
+				:style="{ justifyContent: props.justifyButtons }"
```
Comment (alexgrozav) [inline packages/design-system/src/components/N8nTooltip/Tooltip.vue:45]:
```suggestion
				:style="{ justifyContent: justifyButtons }"
```

File: packages/design-system/src/components/N8nTooltip/Tooltip.vue [hunk 4]
```diff
@@ -48,15 +37,15 @@ export default defineComponent({
...
 			>
 				<N8nButton
-					v-for="button in buttons"
+					v-for="button in props.buttons"
```
Comment (alexgrozav) [inline packages/design-system/src/components/N8nTooltip/Tooltip.vue:48]:
```suggestion
					v-for="button in buttons"
```

File: packages/design-system/src/components/N8nTooltip/Tooltip.vue [hunk 5]
```diff
@@ -1,44 +1,33 @@
...
+	| 'space-evenly';
+
+const props = defineProps({
+	...ElTooltip.props,
+	content: {
+		type: String,
+		default: '',
+	},
+	justifyButtons: {
+		type: String as () => Justify,
```
Comment (alexgrozav) [inline packages/design-system/src/components/N8nTooltip/Tooltip.vue:25]: Use `PropType` for defining custom prop types:

```suggestion
		type: String as PropType<Justify>,
```

File: packages/design-system/src/components/N8nTooltip/Tooltip.vue [hunk 6]
```diff
@@ -1,44 +1,33 @@
...
-					'space-around',
-					'space-evenly',
-				].includes(value),
-		},
-		buttons: {
-			type: Array as PropType<IN8nButton[]>,
-			default: () => [],
-		},
+	buttons: {
+		type: Array as () => IN8nButton[],
```
Comment (alexgrozav) [inline packages/design-system/src/components/N8nTooltip/Tooltip.vue:29]:
```suggestion
		type: Array as PropType<IN8nButton[]>,
```

**Review by alexgrozav — DISMISSED** [review]
Left a few small remarks. Thanks for addressing this.

File: packages/design-system/src/components/N8nTooltip/Tooltip.vue [hunk 7]
```diff
@@ -1,44 +1,33 @@
...
+	...ElTooltip.props,
+	content: {
+		type: String,
+		default: '',
+	},
+	justifyButtons: {
+		type: String as () => Justify,
+		default: 'flex-end',
 	},
-	inheritAttrs: false,
```
Comment (alexgrozav) [inline packages/design-system/src/components/N8nTooltip/Tooltip.vue:14]: Forgot this part. You can use `defineOptions` to add component options meta attributes.

```
defineOptions({ inheritAttrs: false })

File: packages/design-system/src/components/N8nTooltip/Tooltip.vue (hunk 1 above)
Comment (tomi) [reply packages/design-system/src/components/N8nTooltip/Tooltip.vue:40]: If I remove the `props` it complains about the properties not existing:

image

It seems extending the `ElTooltip.props` messes up the props. Any suggestion what's the correct way to extend another components props with the composition API?

**Review by elsmr — APPROVED** [review]

**Comment by Joffcom** [comment]
Got released with [`n8n@1.59.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.59.0)

**Pull request merged**